File: met-api/src/met_api/schemas/submission_version.py
"""Submission Version schema class.

Manages the submission version serialization.
"""

from marshmallow import EXCLUDE, Schema, fields


class SubmissionVersionSchema(Schema):
    """Schema for submission version."""

    class Meta:  # pylint: disable=too-few-public-methods
        """Exclude unknown fields in the deserialized output."""

        unknown = EXCLUDE

    id = fields.Int(data_key='id')
    submission_id = fields.Int(data_key='submission_id')
    version_number = fields.Int(data_key='version_number')
    comment_json = fields.List(fields.Dict(), data_key='comment_json')
    comment_status_id = fields.Int(data_key='comment_status_id')
    reviewed_by = fields.Str(data_key='reviewed_by')
    review_date = fields.Str(data_key='review_date')
    has_personal_info = fields.Bool(data_key='has_personal_info')
    has_profanity = fields.Bool(data_key='has_profanity')
    has_threat = fields.Bool(data_key='has_threat')
    rejected_reason_other = fields.Str(data_key='rejected_reason_other')
    notify_email = fields.Bool(data_key='notify_email')
    staff_note_json = fields.List(fields.Dict(), data_key='staff_note_json')
    submission_json = fields.Dict(data_key='submission_json')
    created_date = fields.Str(data_key='created_date')

File: met-web/src/models/submissionVersion.ts
export interface SubmissionVersion {
    id: number;
    submission_id: number;
    version_number: number;
    comment_json: VersionComment[];
    comment_status_id: number;
    reviewed_by: string;
    review_date: string;
    has_personal_info: boolean;
    has_profanity: boolean;
    has_threat: boolean;
    rejected_reason_other: string;
    notify_email: boolean;
    staff_note_json: VersionStaffNote[];
    submission_json: unknown;
    created_date: string;
}

export interface VersionComment {
    id: number;
    text: string;
    component_id: string;
    submission_date: string | null;
}

export interface VersionStaffNote {
    id: number;
    note: string;
    note_type: string;
}
